### PR TITLE
Add a flag for Marking PlanDefintion a template 

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-server-core/issues/222
https://github.com/OpenSRP/opensrp-server-web/issues/373

This PR comprises of adding a field in PlanDefinition and returning plans or plan templates based on the request parameter.

### CODE REVIEW

**Comment by rehammuzzamil** [comment]
> You should modify existing methods on `PlanRepository` to accept a boolean param that controls is templates are being returned on not. We don't need to add new methods on repository or service

Okay noted, Thanks. But I think we need to make a new method on the repository which takes a Boolean parameter and returns results based on that parameter. I believe, we can not modify the existing method getAll() to take Boolean param as its defined in BaseRepository. @githengi

File: src/main/java/org/opensrp/domain/postgres/PlanExample.java [hunk 1]
```diff
@@ -174,23 +174,23 @@ public boolean isValid() {
 
 		protected void addCriterion(String condition) {
 			if (condition == null) {
-				throw new IllegalArgumentException("Value for condition cannot be null");
+				throw new RuntimeException("Value for condition cannot be null");
```
Comment (githengi) [inline src/main/java/org/opensrp/domain/postgres/PlanExample.java:177]: Change all `RuntimeException`  to `IllegalArgumentException`

**Comment by githengi** [comment]
On planRepository and PlanService add a boolean variable to the methods returning a list of plans, this will allow adding an optional param to the API with a default of `false`

File: src/main/java/org/opensrp/domain/postgres/PlanExample.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/domain/postgres/PlanExample.java]: Done

**Comment by rehammuzzamil** [comment]
> On planRepository and PlanService add a boolean variable to the methods returning a list of plans, this will allow adding an optional param to the API with a default of `false`

Added boolean parameter in plan service and plan repo methods which are returning the list of PlanDefinition

**Comment by githengi** [comment]
bump version

**Comment by rehammuzzamil** [comment]
> Fix Unit tests that are broken

Done

**Comment by rehammuzzamil** [comment]
> > On planRepository and PlanService add a boolean variable to the methods returning a list of plans, this will allow adding an optional param to the API with a default of `false`
> 
> Added boolean parameter in plan service and plan repo methods which are returning the list of PlanDefinition

I have added boolean param and provided its complete implementation i.e. setting criteria of **andExperimentalEqualTo** only in **getAllPlans(boolean experimental)** method.  In other methods, I have only added boolean param but did not make a change in the criteria as other APIs implementation change is not under the scope of this issue. Let me know if I should add the criteria of experimentalEqualTo in other methods as well. @githengi

**Pull request merged**